packages/forms/src/UIForm/Widget/condition.js

```diff
@@ -1,29 +1,172 @@
 import get from 'lodash/get';
-import includes from 'lodash/includes';
+import findIndex from 'lodash/findIndex';
 
-function getValue(value, strategy) {
-	if (!strategy) {
+/**
+ * Tries to ensure the returned value is a number.
+ * It supports string and number for now.
+ */
+function stringOrNumberToNumber(value) {
+	if (typeof value === 'number') {
+		return value;
+	}
+	if (typeof value === 'string') {
+		return Number(value);
+	}
+	const error = { error: 'the passed value is not a string or a number', value };
+	throw error;
+}
+
+/**
+ * Tries to make the returned value a string.
+ * It supports number and string as input.
+ * In case of an error it will call onError. By default it throws an exception.
+ */
+function stringOrNumberToString(
+	value,
+	onError = val => {
+		const error = { error: 'the passed value is not a string or a number', value: val };
+		throw error;
+	},
+) {
+	if (!value) {
 		return value;
 	}
-	switch (strategy.toLowerCase()) {
-		case 'length':
-			if (value && value.length) {
-				return value.length;
+	if (typeof value === 'number') {
+		return JSON.stringify(value);
+	}
+	if (typeof value === 'string') {
+		return value;
+	}
+	return onError(value);
+}
+
+/**
+ * "foo=bar;dummy=1" will return [{foo:"bar",dummy:"1"}]
+ */
+function parseParameters(parameterString) {
+	return parameterString
+		.split(';')
+		.map(it => it.trim())
+		.filter(it => it.length > 0)
+		.map(it => {
+			const sep = it.indexOf('=');
+			if (sep > 0) {
+				return {
+					[it.substring(0, sep).trim()]: it.substring(sep + 1, it.length).trim(),
+				};
 			}
-			return 0;
-		default:
-			return false;
+			return {
+				value: it.trim(),
+			};
+		})
+		.reduce(
+			(a, v) => ({
+				...a,
+				...v,
+			}),
+			{},
+		);
+}
+
+/**
+ * parses a strategy based on this pattern "name(parameters)" and returns
+ * an object containing { name, params }.
+ */
+function parseStrategy(strategy) {
+	if (!strategy) {
+		return strategy;
 	}
+
+	const strategyPattern = /^([a-zA-Z]{1}\w*)(?:\((.+)?\))?$/;
+	const matches = strategy.match(strategyPattern);
+	const isMatching = matches !== undefined;
+
+	const name = isMatching ? matches[1] : strategy;
+	const parameters = isMatching ? matches[2] : undefined;
+
+	const hasParameters = parameters !== undefined;
+
+	return {
+		name: name.toLowerCase(),
+		params: hasParameters ? parseParameters(parameters) : {},
+	};
 }
 
+function areEqualsAsNumbers(expected, actualStringOrNumber) {
+	return expected === stringOrNumberToNumber(actualStringOrNumber);
+}
+
+function containsString(expected, actualStringOrNumber, valueProcessor = v => v) {
```

Comment:
i think a function prefixed by contains should always return a boolean

Reply:
do you mean you want a !! at the beginning?